### PR TITLE
Station Locked firing pins

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request

New firing that only work well not on station.

## Why It's Good For The Game

Now you can defang space loot, space raiders and miners if they start doing some sus stuff

## Changelog
:cl:
add: Firing pins that only works when not on station
/:cl:

### CODE REVIEW

**Comment by Tupinambis** [comment]
Genuinely a good concept honestly. PERFECT for away missions (and might make away missions something that can occur more often?)

File: code/modules/projectiles/pins.dm [hunk 1]
```diff
@@ -229,3 +226,16 @@
...
 		gun.pin = null
 	return ..()
+
+//Station Locked
+
+/obj/item/firing_pin/away
+	name = "station locked pin"
+	desc = "A firing pin that only will fire when off the station."
+
+/obj/item/firing_pin/away/pin_auth(mob/living/user)
```
Comment (silicons) [inline code/modules/projectiles/pins.dm:236]: make sure it respects emagging if it doesn't already, just checking.

**Comment by YakumoChen** [comment]
Away missions only actually happen on extended shifts, which usually feature relaxed security overall, they worry less about guns in general (to forget the point most away missions tend to be packed to the gills with guns regardless)

Unless you're using this to space explore (good luck finding anything first if you're waiting for weapons research to be done) I don't see any actual value of this PR or item. It doesn't even stop use on the station mining areas (not that any gun besides the PKA has value on the mine anyway)

File: code/modules/projectiles/pins.dm (hunk 1 above)
Comment (Trilbyspaceclone) [reply code/modules/projectiles/pins.dm:236]: It does, or should like any normal emagged firing pin.

File: code/modules/projectiles/pins.dm [hunk 2]
```diff
@@ -229,3 +226,16 @@
...
+	var/station_area = get_area(src)
+	if(station_area in GLOB.the_station_areas)
```
Comment (Ghommie) [inline code/modules/projectiles/pins.dm]:
```suggestion
	if(!station_area || is_station_level(station_area.z))
```

**Review by Ghommie — CHANGES_REQUESTED** [review]
For reference: https://github.com/tgstation/tgstation/blob/bf2d1b82fd322e253bc307c792ac31c8b1ce291f/code/modules/projectiles/pins.dm#L299

File: code/modules/projectiles/pins.dm [hunk 3]
```diff
@@ -235,7 +235,7 @@
 
 /obj/item/firing_pin/away/pin_auth(mob/living/user)
 	var/station_area = get_area(src)
```
Comment (Ghommie) [inline code/modules/projectiles/pins.dm]:
```suggestion
	var/area/station_area = get_area(src)
```

**Review by Ghommie — APPROVED** [review]

**Pull request merged**